### DISCUSSION TITLE
Battery life reports

### DISCUSSION THREAD

**Discussion by pmithrandir** [OpenSeizureDetector/Android_Pebble_SD#134] [04_Devices]
Hello,

I would like to list user feedback on different watch to measure the average battery life.
I expect to be able to either find some watch that are less efficient, or maybe to help user to identify that their device has a lower battery life than other similar devices.

I have a Vivoactive 4 running the watch app 1.4.1n.
Battery life running in low data mode is around 12 hours. 7-8 hours in normal data mode.
It's barely usable for normal usage, usually doesn't make a full school day, not also a night.

And on you side ? Any similar measure ?

Pierre

**Comment by jones139** [comment]
For our 'Production' set-up we use an old VivoactiveHR, which gives us about 24 hours battery life.

**Comment by AroonPro** [comment]
I have asked Bing for power drainage, it is giving me results and an answer, that this type of watch may have flashlight activated or reading (I presume non-stop live) blood oxygen or have live GPS tracking on. Lots of Garmin users are affected.

My Samsung Watch 4 is also suseptible to power drainage.

My power drainage also took place when the app requested that the cpu did not go to sleep.  Now I have added partially to wake up on sensor events or on battery changes. 

Memory efficiency and battery health now these days are difficult because simply said BloatWare is being provided to pay the brand.


I suggest to include gps location (activate receiver) only when seizure detected. As I don't have Garmin, I cannot say directly where to resolve. Only feel in.

**Reply by pmithrandir** [reply]
Hello,
I checked and I don't think the GPS is enabled on the garmin watch when using the OSD "activity". Or if it is, it doesn't provides the parameters present for other activities.
From my point of iew, GPS is not useful at all because the watch in itself doesn't send the alarm, the phone does. Knowing that, the GPS from the phone could be used more efficiently.

If you have more info / lead on the subject, please correct me.

**Reply by jones139** [reply]
Hi,
The OSD app should not be switching on the GPS on the watch - we do not use it - we use the phone's GPS location instead.
It is very odd how some watches have much longer battery life than others - I did wonder if it might be the screen redrawing, but when I tried changing the screen refresh from 1 sec to  5 sec period, it did not make an appreciable difference.   The Low Data mode in V1.4.x sends only about 25% of the data compared to normal, but I think you have already tried that?
The source code for the V1.4.x version of the watch app is here if you'd like to see if you can spot anything that we could do to make it more efficient:  https://github.com/OpenSeizureDetector/Garmin_SD/tree/V1.4.x/source

**Reply by pmithrandir** [reply]
Hi,

I'm wondering if that line : https://github.com/OpenSeizureDetector/Garmin_SD/blob/c5a589caf59b0f97c3b033a502b25d2b75dfe56b/source/GarminSDDataHandler.mc#L214 could activate the O2, even if I disabled it in the garmin watch main setting. 

It displays O2 for my son always, so I think it might drain the battery(as suggested in the comment).

I could suggest a plan is 2 steps : 
 - remove that condition, and rebuild the app. 
 - I test it.

If battery life is improved, I would suggest adding a new setting in garmin app to enable/disable the O2 sensor + add the condition in the above line.
If not, to never release that change.

What do you think ?

**Reply by jones139** [reply]
Well spotted - I must have found that the battery life was not a problem when I was testing it, so left that line in - but both of my devices will hardly ever do an O2 measurement - if yours is actually trying to do it continuously that might be the issue.
I'll get you a test version tomorrow (I'm having trouble with the compiler just now), then we can see what effect it has.
Thanks!

Graham.

**Reply by jones139** [reply]
I have built a test version with the o2sat sensor disabled as you suggested here:  https://github.com/OpenSeizureDetector/Garmin_SD/blob/V1.4.x/build/GarminSD_v1.4.1_no_o2sat.prg

Please will you try this and see if it has any effect?
Note that it still attempts to read the sensor, but I think it will just return zero if it is not enabled- if  there is no battery effect, I'll try inhibiting that too.

**Reply by pmithrandir** [reply]
Hi,

I installed it. I'll test it in the next few days. We should have quickly an answer.

Regards,
Pierre

**Reply by pmithrandir** [reply]
Hi.
It seems better, but not stunning.
I was around 7-8 hours, looks like we are now above 10-11 hours. (From 8am to 6pm, with 9% battery remaining).

I'll confirm it over the next days.

**Comment by jones139** [comment]
Thanks - it sounds better, but as you say, still not really good enough.
 I'll do a proper update to it with enabling the O2 sensor as an option in
the menu.

On Tue, 30 Jan 2024 at 20:05, Pierre Bonneau ***@***.***>
wrote:

> Hi.
> It seems better, but not stunning.
> I was around 7-8 hours, looks like we are now above 10-11 hours. (From 8am
> to 6pm, with 9% battery remaining).
>
> I'll confirm it over the next days.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OpenSeizureDetector/discussions/134#discussioncomment-8310869>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACLSY6EFBRBHMBPPKB4PC3YRFHCPAVCNFSM6AAAAABCLDZMT2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGMJQHA3DS>
> .
> You are receiving this because you commented.Message ID:
> <OpenSeizureDetector/Android_Pebble_SD/repo-discussions/134/comments/8310869
> @github.com>
>


-- 
Graham Jones <http://google.com/+GrahamJones>
Hartlepool, UK.

**Reply by pmithrandir** [reply]
I think I did it on my watch. Waiting to test it tomorrow.

**Comment by pmithrandir** [comment]
With version 2.X, I now reach 15 hours of autonomy with Vivoactive 4

Interesting point is the distance between watch and phone.
2 meters : 15 hours
4 meters (and a wall) 13 hours
7 meters: 11-12 hours

If you use it by night to watch the sleep, I would recommend to keep the phone close to the bed and to use the network mode on your own phone.

**Reply by jones139** [reply]
That is very useful to know, thank you - it must mean that BLE varies the transmitter power to achieve the desired signal to noise ratio?

This could help with the issue we are seeing with pinetime range being lower than you might expect - I will see if it is being optimised for battery life rather than performance.

**Comment by pmithrandir** [comment]
I bought a forerunner 255 the other day (second hand)

After a 12 hours night, the battery is around 89%.

It's not even comparable to the vivoactive v4 that was almost empty by that time.

And bracelet is compatible to wrist from 11cm, ok for a kid.
